docs/index/secondary-index-guide.md

```diff
@@ -0,0 +1,192 @@
+first hit the corresponding SI table and give input to main table for further pruning.
+
+
+For the main table **sales** and SI table  **index_sales** created above, following queries
+```
+SELECT country, sex from sales where user_id = 'xxx'
+
+SELECT country, sex from sales where user_id = 'xxx' and country = 'INDIA'
+```
+
+will be transformed by CarbonData's CarbonSITransformationRule to query against SI table
+**index_sales** first which will be input to the main table **sales**
+
+
+## Loading data
+
+### Loading data to Secondary Index table(s).
+
+*case1:* When SI table is created and the main table does not have any data. In this case every
+consecutive load will load to SI table once main table data load is finished.
+
+*case2:* When SI table is created and main table already contains some data, then SI creation will
+also load to SI table with same number of segments as main table. There after, consecutive load to
+main table will load to SI table also.
+
+ **NOTE**:
+ * In case of any failure to SI table, then we make the SI table disable by setting a hive serde
+ property. The subsequent main table load will load the old failed loads along with current load and
+ makes the SI table enable and available for query.
+
+## Querying data
+Direct query can be made on SI tables to see the data present in position reference columns.
+When a filter query is fired, if the filter column is a secondary index column, then plan is
+transformed accordingly to hit SI table first to make better pruning with main table and in turn
+helps for faster query results.
+
+User can verify whether a query can leverage SI table or not by executing `EXPLAIN`
+command, which will show the transformed logical plan, and thus user can check whether SI table
+table is selected.
+
+
+## Compacting SI table
+
+### Compacting SI table table through Main Table compaction
+Running Compaction command (`ALTER TABLE COMPACT`)[COMPACTION TYPE-> MINOR/MAJOR] on main table will
+automatically delete all the old segments of SI and creates a new segment with same name as main
+table compacted segmet and loads data to it.
+
+### Compacting SI table's individual segment(s) through REBUILD command
+Where there are so many small files present in the SI table, then we can use REBUILD command to
+compact the files within an SI segment to avoid many small files.
+
+  ```
+  REBUILD INDEX sales_index
+  ```
+This command merges data files in  each segment of SI table.
+
+  ```
+  REBUILD INDEX sales_index WHERE SEGMENT.ID IN(1)
+  ```
+This command merges data files within specified segment of SI table.
+
+## How to skip Secondary Index?
+When Secondary indexes are created on a table(s), always data fetching happens from secondary
+indexes created on the maintables for better performance. But sometimes, data fetching from the
+secondary index might degrade query performance. So to avoid such secondary indexes, we use NI as a
```

Comment:
can mention when it will be degraded.
For example, when the data is sparse and most of the blocklet need to be scanned.